### PR TITLE
[Edge] Add mqtt hybrid unittest @open sesame 08/11 20:47

### PR DESCRIPTION
**Pull request by gichan-jang**
- Add mqtt hybrid unittest.

Signed-off-by: gichan <gichan2.jang@samsung.com>

### CODE REVIEW

File: src/libnnstreamer-edge/nnstreamer-edge-mqtt.c [hunk 1]
```diff
@@ -496,7 +497,7 @@ nns_edge_mqtt_get_message (nns_edge_h edge_h, char **msg)
 
   eh = (nns_edge_handle_s *) edge_h;
 
-  if (!NNS_EDGE_MAGIC_IS_VALID (eh) || !eh->broker_h) {
+  if (!NNS_EDGE_MAGIC_IS_VALID (eh) || !eh->broker_h || !msg) {
```
Comment (jaeyun-jung) [inline src/libnnstreamer-edge/nnstreamer-edge-mqtt.c:500]: can you separate this condition (!msg) to print exact error message?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # nnstreamer-edge test
+SET(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -DENABLE_MQTT=1")
```
Comment (jaeyun-jung) [inline tests/CMakeLists.txt:2]: let's move this to root make file, also need fix this.
`SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DENABLE_MQTT=1")`

File: tests/unittest_nnstreamer-edge.cc [hunk 3]
```diff
@@ -2815,6 +2815,460 @@ TEST(edgeMeta, deserializeInvalidParam03_n)
...
+ * @brief Check whether MQTT broker is running or not.
+ */
+static int
+_check_mqtt_broker ()
+{
+  char *check_broker_cmd;
+  int ret = 0;
+
+  check_broker_cmd = nns_edge_strdup ("ps aux | grep mosquitto | grep -v grep");
+  ret = system (check_broker_cmd);
```
Comment (jaeyun-jung) [inline tests/unittest_nnstreamer-edge.cc:2893]: IMO check_broker_cmd is unnecessary.
ret = system ("ps aux | grep mosquitto | grep -v grep");

File: tests/unittest_nnstreamer-edge.cc [hunk 4]
```diff
@@ -2815,6 +2815,460 @@ TEST(edgeMeta, deserializeInvalidParam03_n)
...
+  /* Wait for server/client thread */
+  do {
+    usleep (20000);
+  } while (!g_main_loop_is_running (_td_server->loop));
+
+  do {
+    usleep (20000);
+  } while (!g_main_loop_is_running (_td_client->loop));
+
+  ret = nns_edge_connect (client_h, "tcp://localhost", 1883);
```
Comment (jaeyun-jung) [inline tests/unittest_nnstreamer-edge.cc:2960]: line 2930 and 2960, how about getting available port using nns_edge_get_available_port()?

File: tests/CMakeLists.txt (hunk 2 above)
Comment (gichan-jang) [reply tests/CMakeLists.txt]: Updated. Thanks.

File: tests/unittest_nnstreamer-edge.cc (hunk 3 above)
Comment (gichan-jang) [reply tests/unittest_nnstreamer-edge.cc]: Good! Updated.

File: tests/unittest_nnstreamer-edge.cc (hunk 4 above)
Comment (gichan-jang) [reply tests/unittest_nnstreamer-edge.cc:2958]: This is for mosquitto broker port. Need to set 1883.

File: src/libnnstreamer-edge/nnstreamer-edge-mqtt.c (hunk 1 above)
Comment (gichan-jang) [reply src/libnnstreamer-edge/nnstreamer-edge-mqtt.c]: Updated!

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**